user-script-register/popup/user-script.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const hostsInput =  document.querySelector("#hosts");
+const codeInput =  document.querySelector("#code");
+const lastErrorEl =  document.querySelector("#lastError");
+const lastResultEl =  document.querySelector("#lastResult");
+
+const defaultHosts = "*://*.org/*";
+const defaultCode = `(async function () {
+  console.log("USER SCRIPT EXECUTING on", window.location.href, {
+    documentReadyState: document.readyState,
+  });
+  const oldStoredValue = await GM_getValue("testkey");
+  await GM_setValue("testkey", window.location.href);
+  const newStoredValue = await GM_getValue("testkey");
+
+  const overwriteBody = () => {
+    document.body.innerHTML = \`<h1>This page has been eaten: $\{JSON.stringify({oldStoredValue, newStoredValue})}<h1>\`
+  }
+
+  if (document.body) {
+    overwriteBody();
+  } else {
+    window.addEventListener("load", overwriteBody, {once: true});
+  }
+})();`;
+
+hostsInput.value = defaultHosts;
+codeInput.value = defaultCode;
+
+async function loadLastSetValues() {
+  const params = await browser.storage.local.get();
+
+  const {
+    hosts,
+    code,
+  } = params.lastSetValues || {};
+
+  hostsInput.value = hosts ? hosts.join(",") : defaultHosts;
+  codeInput.value = code ? code : defaultCode;
+
+  lastErrorEl.textContent = params.lastError || "";
+}
+
+function stringToArray(value) {
+  const res = value.split(",").map(el => el.trim()).filter(el => el !== "");
+
+  return res.length > 0 ? res : null;
+}
+
+function stringToBool(value) {
```

Comment:
I think this is now unused and can be deleted?

Reply:
Thanks for spotting that one, removed